app/lib/actions/proxy_action.rb

```diff
@@ -5,6 +5,9 @@ class ProxyAction < Base
 
     middleware.use ::Actions::Middleware::HideSecrets
 
+    execution_plan_hooks.use :clean_remote_task, :on => :stopped
```

Comment:
We should bump dependency on on dynflow to 0.8.37 to ensure we have the hooks in dynflow

Reply:
Done